src/main/java/com/leverx/mediator/repository/destination/impl/CatDestinationRepositoryImpl.java

```diff
@@ -0,0 +1,108 @@
+package com.leverx.mediator.repository.destination.impl;
+
+import static java.util.Arrays.asList;
+
+import static com.leverx.mediator.repository.destination.creator.DestinationCreator.HTTP_CLIENT;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.util.EntityUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Repository;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.leverx.mediator.dto.request.CatRequest;
+import com.leverx.mediator.dto.response.CatResponse;
+import com.leverx.mediator.exception.RepositoryException;
+import com.leverx.mediator.repository.CatRepository;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/** @author Andrei Yahorau */
+@Repository
+@Primary
+@RequiredArgsConstructor
+@Slf4j
+public class CatDestinationRepositoryImpl implements CatRepository {
+
+  private final ObjectMapper mapper;
+
+  @Value("${sap.link.cats}")
+  private final String catsLink;
+
+  @Override
+  public CatResponse save(final CatRequest cat) {
+    log.info("CatRepository. Save cat: {}", cat);
+
+    CatResponse catResponse;
+
+    HttpPost post = new HttpPost(catsLink);
+    try {
+      post.setEntity(new StringEntity(mapper.writeValueAsString(cat)));
+      post.setHeader("Accept", "application/json");
+      post.setHeader("Content-type", "application/json");
```

Comment:
To many work in service methods. Try to decouple logic of configuration http client and business logic.